gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/FlowCatalog.java

```diff
@@ -366,11 +376,17 @@ public Spec getSpecWrapper(URI uri) {
     return responseMap;
   }
 
-  public static boolean isCompileSuccessful(Map<String, AddSpecResponse> responseMap) {
-    AddSpecResponse<String> addSpecResponse = responseMap.getOrDefault(
-        ServiceConfigKeys.GOBBLIN_SERVICE_JOB_SCHEDULER_LISTENER_CLASS, new AddSpecResponse<>(""));
-
-    return isCompileSuccessful(addSpecResponse.getValue());
+  public boolean isCompileSuccessful(Map<String, AddSpecResponse> responseMap) {
+    for (SpecCatalogListener listener: this.listeners.getListeners()) {
+      String listenerClass = listener.getClass().getCanonicalName();
+      AddSpecResponse<String> addSpecResponse =
```

Comment:
Is it safe to call listeners from this place? I wonder if they have any side effects, and will call them multiple times for the same flow. It would be good to review what  they actually do and what is the contract/docs for the listener interface.

Reply:
I needed this originally for testing purposes but realized that I can mock the name of the expected listener instead